frontend/src/pages/IstioConfigNew/GatewayForm/ListenerList.tsx

```diff
@@ -144,16 +151,28 @@ export const ListenerList: React.FC<ListenerListProps> = (props: ListenerListPro
     if (listenerForm.port.length === 0 || isNaN(Number(listenerForm.port))) return;
     if (listenerForm.hostname.length === 0) return;
 
-    const selector = addSelectorLabels(listenerForm.sSelectorLabels) || {};
+    const selector = addSelectorLabels(listenerForm.sSelectorLabels)[1] || {};
 
     const listener: Listener = {
```

Comment:
I don't understand this part of the code... It seems that we are creating twice the listener yaml... once here and the same in `WizardActions.buildK8sGateway` method.

IMO I would create the yaml here and then in WizardActions just map the listeners as is:
```
    spec: {
      // Default for istio scenarios, user may change it editing YAML
      gatewayClassName: state.gatewayClass,
      listeners: state.listeners
    }
  };

  if (state.addresses.length > 0) {
    k8sGateway.spec.addresses = state.addresses;
  }
```

Reply:
good point